PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/mysql/creation/MySQLCreationAdvanced.java

```diff
@@ -0,0 +1,207 @@
+    }
+
+    private void initListeners() {
+        this.subscriptionComboBox.addItemListener(this::onSubscriptionChanged);
+        this.resourceGroupComboBox.addItemListener(this::onResourceGroupChanged);
+        this.adminUsernameTextField.getDocument().addDocumentListener(generateAdminUsernameListener());
+        this.security.getAllowAccessFromAzureServicesCheckBox().addItemListener(this::onSecurityAllowAccessFromAzureServicesCheckBoxChanged);
+        this.security.getAllowAccessFromLocalMachineCheckBox().addItemListener(this::onSecurityAllowAccessFromLocalMachineCheckBoxChanged);
+    }
+
+    private void onSubscriptionChanged(final ItemEvent e) {
+        if (e.getStateChange() == ItemEvent.SELECTED && e.getItem() instanceof Subscription) {
+            final Subscription subscription = (Subscription) e.getItem();
+            this.resourceGroupComboBox.setSubscription(subscription);
+            this.serverNameTextField.setSubscription(subscription);
+        }
+    }
+
+    private void onResourceGroupChanged(final ItemEvent e) {
+        if (e.getStateChange() == ItemEvent.SELECTED && e.getItem() instanceof ResourceGroup) {
+            final ResourceGroup resourceGroup = (ResourceGroup) e.getItem();
+            this.serverNameTextField.setResourceGroup(resourceGroup);
+        }
+    }
+
+    private DocumentListener generateAdminUsernameListener() {
+        return new TextDocumentListenerAdapter() {
+            @Override
+            public void onDocumentChanged() {
+                if (!adminUsernameTextField.isPasswordInitialized()) {
+                    adminUsernameTextField.setPasswordInitialized(true);
+                }
+            }
+        };
+    }
+
+    private void onSecurityAllowAccessFromAzureServicesCheckBoxChanged(final ItemEvent e) {
+        if (e.getStateChange() == ItemEvent.SELECTED) {
+            config.setAllowAccessFromAzureServices(true);
+        } else if (e.getStateChange() == ItemEvent.DESELECTED) {
+            config.setAllowAccessFromAzureServices(false);
+        }
+    }
+
+    private void onSecurityAllowAccessFromLocalMachineCheckBoxChanged(final ItemEvent e) {
+        if (e.getStateChange() == ItemEvent.SELECTED) {
+            config.setAllowAccessFromLocalMachine(true);
+        } else if (e.getStateChange() == ItemEvent.DESELECTED) {
+            config.setAllowAccessFromLocalMachine(false);
+        }
+    }
+
+    @Override
+    public void setVisible(boolean visible) {
+        super.setVisible(visible);
+        rootPanel.setVisible(visible);
+    }
+
+    @Override
+    public AzureMySQLConfig getData() {
+        config.setServerName(serverNameTextField.getText());
+        config.setAdminUsername(adminUsernameTextField.getText());
+        config.setPassword(passwordField.getPassword());
+        config.setConfirmPassword(confirmPasswordField.getPassword());
```

Comment:
IMO, `confirmPassword` should not be part of `AzureMySQLConfig`

Reply:
yes. But let it here should be useful. we need to keep to copy during switching between basic and advanced mode.

Reply:
I would prefer required user to re-input the confirmation password or do something specifically for it instead of make it part of our biz POJO

Reply:
how about not kink on this. as I think, we can regard XXXConfig as a collection POJO of form inputs data of a dialog. it is not a strict business value.